lib/web/apps.go

```diff
@@ -174,18 +175,20 @@ func (h *Handler) createAppSession(w http.ResponseWriter, r *http.Request, p htt
 	}, nil
 }
 
-// waitForSession will block until the requested session shows up in the
+// waitForAppSession will block until the requested application session shows up in the
 // cache or a timeout occurs.
-func (h *Handler) waitForSession(ctx context.Context, sessionID string) error {
-	timeout := time.NewTimer(defaults.WebHeadersTimeout)
-	defer timeout.Stop()
-
+func (h *Handler) waitForAppSession(ctx context.Context, sessionID string) error {
+	_, err := h.cfg.AccessPoint.GetAppSession(ctx, services.GetAppSessionRequest{SessionID: sessionID})
+	if err == nil {
```

Comment:
Handle error other than not found?

Reply:
I've added the log message similar to web session wait.